python/pyspark/errors/error_classes.py

```diff
@@ -39,6 +39,11 @@
       "Function `<func_name>` should return Column, got <return_type>."
     ]
   },
+  "IS_NOT_ITERABLE" : {
```

Comment:
Nit: `IS_` seems to be redundant. Other message names do not adopt `IS_`.

Reply:
I intentionally added `IS_` since the error message format is different from other `NOT_***` error classes, e.g.

```
Argument `<arg_name>` should be a ** or **, got <arg_type>."
```

But I also agree with it looks a bit redundant compare to others.

Let me just use `NOT_ITERABLE` anyway.

Thanks! :-)